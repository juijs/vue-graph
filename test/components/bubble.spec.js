import { mount } from '@vue/test-utils'
import GraphBubble from '@/components/bubble'

describe('bubble.js', () => {
    it('snapshot', () => {
        const wrapper = mount(GraphBubble, {
            props: {
                width: 600,
                height: 400,
                axisMin: 0,
                axisMax: 50,
                axisInterval: 1000 * 60 * 60,
                axisFormat: 'HH',
                showText: true,
                labels: [ new Date('2018-07-09 00:00:00'), new Date('2018-07-10 00:00:00') ],
                values: [
                    [ new Date('2018-07-09 05:23:00'), 20 ],
                    [ new Date('2018-07-09 12:00:00'), 15 ],
                    [ new Date('2018-07-09 15:55:00'), 12 ],
                    [ new Date('2018-07-09 19:01:00'), 30 ]
                ]
            }
        });

        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
});