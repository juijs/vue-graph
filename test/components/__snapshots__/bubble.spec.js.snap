// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bubble.js snapshot 1`] = `
<div
  style="position: relative; user-select: none; webkit-user-select: none;"
  unselectable="on"
>
  <svg
    buffered-rendering="dynamic"
    height="400"
    style="font-family:arial,Tahoma,verdana,classic;background:#fff"
    width="600"
  >
    <g
      transform="translate(0.5,0.5)"
    >
      <defs>
        <clippath
          id="axis-clip-id-0.0"
        >
          <rect
            height="300"
            width="500"
            x="0"
            y="0"
          />
        </clippath>
        <clippath
          id="axis-clip-rect-id-0"
        >
          <rect
            height="300"
            width="500"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <rect
        fill="#fff"
        fill-opacity="0"
        height="300"
        rx="0"
        ry="0"
        stroke="#fff"
        stroke-width="0"
        transform="translate(50,50)"
        width="500"
        x="0"
        y="0"
      />
      <g
        class="grid-date"
        transform="translate(50,350)"
      >
        <g
          transform="translate(0,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            00
          </text>
        </g>
        <g
          transform="translate(20.833333333333332,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            01
          </text>
        </g>
        <g
          transform="translate(41.666666666666664,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            02
          </text>
        </g>
        <g
          transform="translate(62.5,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            03
          </text>
        </g>
        <g
          transform="translate(83.33333333333333,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            04
          </text>
        </g>
        <g
          transform="translate(104.16666666666667,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            05
          </text>
        </g>
        <g
          transform="translate(125,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            06
          </text>
        </g>
        <g
          transform="translate(145.83333333333334,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            07
          </text>
        </g>
        <g
          transform="translate(166.66666666666666,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            08
          </text>
        </g>
        <g
          transform="translate(187.5,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            09
          </text>
        </g>
        <g
          transform="translate(208.33333333333334,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            10
          </text>
        </g>
        <g
          transform="translate(229.16666666666666,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            11
          </text>
        </g>
        <g
          transform="translate(250,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            12
          </text>
        </g>
        <g
          transform="translate(270.8333333333333,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            13
          </text>
        </g>
        <g
          transform="translate(291.6666666666667,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            14
          </text>
        </g>
        <g
          transform="translate(312.5,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            15
          </text>
        </g>
        <g
          transform="translate(333.3333333333333,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            16
          </text>
        </g>
        <g
          transform="translate(354.1666666666667,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            17
          </text>
        </g>
        <g
          transform="translate(375,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            18
          </text>
        </g>
        <g
          transform="translate(395.8333333333333,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            19
          </text>
        </g>
        <g
          transform="translate(416.6666666666667,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            20
          </text>
        </g>
        <g
          transform="translate(437.5,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            21
          </text>
        </g>
        <g
          transform="translate(458.3333333333333,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            22
          </text>
        </g>
        <g
          transform="translate(479.1666666666667,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            23
          </text>
        </g>
        <g
          transform="translate(500,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="0"
            y1="0"
            y2="3"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="-300"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            transform="rotate(0 0,13)"
            x="0"
            y="13"
          >
            00
          </text>
        </g>
        <line
          stroke="#bfbfbf"
          stroke-opacity="1"
          stroke-width="2"
          x1="0"
          x2="500"
          y1="0"
          y2="0"
        />
      </g>
      <g
        transform="translate(50,50)"
      />
      <g
        class="grid-range"
        transform="translate(50,50)"
      >
        <g
          transform="translate(0,0)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            50
          </text>
        </g>
        <g
          transform="translate(0,30)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            45
          </text>
        </g>
        <g
          transform="translate(0,60)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            40
          </text>
        </g>
        <g
          transform="translate(0,90)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            35
          </text>
        </g>
        <g
          transform="translate(0,120)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            30
          </text>
        </g>
        <g
          transform="translate(0,150)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            25
          </text>
        </g>
        <g
          transform="translate(0,180)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            20
          </text>
        </g>
        <g
          transform="translate(0,210)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            15
          </text>
        </g>
        <g
          transform="translate(0,240)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            10
          </text>
        </g>
        <g
          transform="translate(0,270)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="0"
            x2="500"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            5
          </text>
        </g>
        <g
          transform="translate(0,300)"
        >
          <line
            stroke="#bfbfbf"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1.5"
            x1="0"
            x2="-3"
            y1="0"
            y2="0"
          />
          <text
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            transform="rotate(0 -8,0)"
            x="-8"
            y="0"
          >
            0
          </text>
        </g>
        <line
          stroke="#bfbfbf"
          stroke-opacity="1"
          stroke-width="2"
          x1="0"
          x2="0"
          y1="0"
          y2="300"
        />
      </g>
      <g
        transform="translate(50,50)"
      />
      <g
        class="grid-panel"
        display="none"
        transform="translate(50,50)"
      >
        <rect
          fill="transparent"
          height="300"
          stroke="transparent"
          width="500"
          x="0"
          y="0"
        />
      </g>
      <g
        class="brush-bubble"
        transform="translate(50,50)"
      >
        <g
          transform="translate(112.15277777777779,180)"
        >
          <circle
            fill="#7977C2"
            fill-opacity="0.5"
            r="15"
            stroke="#7977C2"
            stroke-width="1"
          />
          <text
            dy="3"
            fill="#fff"
            font-size="12"
            text-anchor="middle"
          >
            20
          </text>
        </g>
        <g
          transform="translate(250,210)"
        >
          <circle
            fill="#7977C2"
            fill-opacity="0.5"
            r="12.5"
            stroke="#7977C2"
            stroke-width="1"
          />
          <text
            dy="3"
            fill="#fff"
            font-size="12"
            text-anchor="middle"
          >
            15
          </text>
        </g>
        <g
          transform="translate(331.59722222222223,228)"
        >
          <circle
            fill="#7977C2"
            fill-opacity="0.5"
            r="11"
            stroke="#7977C2"
            stroke-width="1"
          />
          <text
            dy="3"
            fill="#fff"
            font-size="12"
            text-anchor="middle"
          >
            12
          </text>
        </g>
        <g
          transform="translate(396.18055555555554,120)"
        >
          <circle
            fill="#7977C2"
            fill-opacity="0.5"
            r="20"
            stroke="#7977C2"
            stroke-width="1"
          />
          <text
            dy="3"
            fill="#fff"
            font-size="12"
            text-anchor="middle"
          >
            30
          </text>
        </g>
      </g>
    </g>
    <g
      transform="translate(0.5,0.5)"
    />
  </svg>
</div>
`;
